src/codegen/expression.rs

```diff
@@ -2238,6 +2250,45 @@ fn expr_builtin(
                 value: 0.into(),
             }
         }
+        ast::Builtin::TypeName => {
+            let ast::Expression::TypeOperator {
+                ty: Type::Contract(no),
+                ..
+            } = &args[0]
+            else {
+                unreachable!();
+            };
+
+            let value = ns.contracts[*no].id.name.as_bytes().to_owned();
+
+            Expression::BytesLiteral {
+                loc: *loc,
+                ty: Type::String,
+                value,
+            }
+        }
+        ast::Builtin::TypeInterfaceId => {
+            let ast::Expression::TypeOperator {
+                ty: Type::Contract(contract_no),
+                ..
+            } = &args[0]
+            else {
+                unreachable!();
+            };
+
+            interfaceid(ns, *contract_no, loc)
+        }
+        ast::Builtin::TypeRuntimeCode | ast::Builtin::TypeCreatorCode => {
+            let ast::Expression::TypeOperator {
+                ty: Type::Contract(contract_no),
+                ..
+            } = &args[0]
```

Comment:
Instead of a builtin, wouldn't it be nicer of item we access from a type operator was a member access? This way you could define the possible members for the operator.

I found it somewhat uncomfortable that you require the argument to be `TypeOperator` in all cases with an unreachable.

Reply:
As there is a constraint for the argument, perhaps it could be enforced in another way that avoids the unreachable.

Reply:
What do you mean by member access? `Expression::StructMember`? That expression is not suitable, it takes a field number and it has "Struct" in its name. 

You may want the location of `type(..)` in `type(int)  .    min` and `type(int)` can be free standing too, so a separate `Expression::TypeOperator` enum makes sense to me.

Please make suggestions because I have no idea.